### ISSUE TITLE
 There was an unexpected error while connecting to the machine - iOS

### ISSUE THREAD

**Issue by Chriscope0** [cockpit-project/cockpit#15106]
Cockpit version: 215
OS: Ubuntu  20.04.1 LTS

1. I have cockpit installed with the VM addon.
2. I can access the GUI via default port 9090 on my windows 10 laptop running Chrome Version 87.0.4280.88.
3. However, I can't access the cockpit GUI using my iOS devices:
- iPad Air gen 3 iOS 14.3
- iPhone SE iOS 14.3
4. I have tried both Chrome and Firefox web browser app.
5. I can access the cockpit login page and can enter my credentials. when I click 'log in' i get the following:
**"Connection Failed There was an unexpected error while connecting to the machine. Messages related to the failure might be found in the journal: journalctl -u cockpit."**

6. I open a terminal to the server, access '`journalctl -u cockpit`'. I go through the process again to get the failure and it logs  
` "server cockpit-tls[1452]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received."`

is there a solution to this problem? I have searched and found a few conflicting concepts. unfortunately, I have a limited understanding with little experience of TLS and certifications.

**Comment by martinpitt** [comment]
At least on Safari it is a very old problem, see issue #10146. It has had a bug where it asks you about an unknown/invalid self-signed certificate on first connection, you can accept it, and then it still refuses to use it for websockets (it only works for https). This is unfortunately hard to work around, [here are some options](https://github.com/cockpit-project/cockpit/issues/10146#issuecomment-660950689).

@larskarlitski was able to make this work by importing Cockpit's ca.crt into the OS as valid TLS certificate. I assume you got the "unknown certificate" page as well? There you can download it.

**Comment by Chriscope0** [comment]
Thank you for your reply, 

Unfortunately, it seems the workarounds are less than ideal. unsecured HTTP seems dangerous for this type of information, even within the confines of a secured local network.

I feel lost with the certifications, I'm researching and getting names of software/service but I'm not sure how these all piece together.

**Comment by rtahiry** [comment]
This issue should be resolved by the workaround in the thread @martinpitt posted above https://github.com/cockpit-project/cockpit/issues/10146#issuecomment-840975805

**Comment by garrett** [comment]
> I have tried both Chrome and Firefox web browser app.

@Chriscope0:

On iPhones and iPads, there is only Safari's WebKit. "Chrome" and "Safari" are just wrappers around the WebKit browser engine as per Apple's restrictions... so every "browser" on iOS will act the same way as Safari when rendering a site.

This bug you're hitting is an Apple specific bug (not a Cockpit one) and Apple finally fixed in iOS 15.x (I think 15.1?); so you'll need to upgrade iOS for the fix. Sadly, Apple does EOL their phones without updating their browser version, so the short of it is that your devices are too old to run Cockpit due to a bug in iOS certificate handling. Thankfully, does upgrade their phones and tablets quite a bit longer than anyone else. (However, even old, unsupported Android can run the latest Chrome and Firefox, so it's a mixed bag, really.)

_However_, **great news for you specifically**: it looks like the iPhone SE 1st generation and the iPad Air 3rd generation both support 15.2, so please upgrade your devices and try using Cockpit again.

- https://support.apple.com/guide/iphone/supported-models-iphe3fa5df43/ios
- https://support.apple.com/guide/ipad/supported-models-ipad213a25b2/ipados

IIRC, I think you don't even need the workaround that was linked in the 15.2. I think you can now just use your device, browse to Cockpit, accept the certificate, and you're good when you're using 15.2. So... please upgrade! (You should upgrade for other reasons too, like device security. And you'll get a bunch of new features as well.)

**Comment by garrett** [comment]
Since it is about an old bug related to an Apple issue on old versions of iOS and iPadOS, not a bug in Cockpit, and has been fixed in recent versions of iOS and iPadOS, I'm going to close it.

@Chriscope0: Please reopen with a comment if you cannot connect to Cockpit after upgrading your devices to 15.2. Otherwise: Good luck! I hope upgrading does indeed fix it for you. (It should.)

**Comment by 0x30c4** [comment]
I tried with firefox on linux but same problem.

**Comment by garrett** [comment]
@0x30c4: This is an older issue specifically about iOS. Please open a new issue and provide details, including which version of Cockpit you are using, what browser you're using, what Linux distribution you're running Cockpit on, what you are trying to do, if you're using a proxy or not.

(FWIW: I develop Cockpit on Linux and access Cockpit from Linux using Firefox and it works perfectly fine. So please be specific when you open your issue. Thanks!)

**Comment by garrett** [comment]
@0x30c4: This is an older issue specifically about iOS. Please open a new issue and provide details, including which version of Cockpit you are using, what browser you're using, what Linux distribution you're running Cockpit on, what you are trying to do, if you're using a proxy or not.

(FWIW: I develop Cockpit on Linux and access Cockpit from Linux using Firefox and it works perfectly fine. So please be specific when you open your issue. Thanks!)

Our new issue form is @ https://github.com/cockpit-project/cockpit/issues/new?assignees=&labels=bug&template=bug_report.yml

**Comment by 0x30c4** [comment]
I was able to fix the issue by changing the ownership of the /run/cockpit
```
chown -R cockpit-ws:cockpit-ws /run/cockpit
```
And adding ```AllowUnencrypted = true``` to the ```/etc/cockpit/cockpit.conf``` file

So, do I need to open a new issue now?

**Comment by garrett** [comment]
> And adding ```AllowUnencrypted = true``` to the ```/etc/cockpit/cockpit.conf``` file

@0x30c4: Cockpit lets you change things as an administrator on your machine. Turning off encryption is not a good idea if it's accessible to the network (even a local network). That's mainly meant for when you're using a proxy to add in TLS separately.

> So, do I need to open a new issue now?

No, your distribution does not use systemd, and [Cockpit needs systemd](https://cockpit-project.org/guide/latest/feature-systemd) to work. Sorry. It's like you're basically trying to use a car that is missing an engine.

I'd recommend switching from Artix to Arch with systemd, as it would be otherwise similar. Or choose [another distribution listed on Cockpit's installation page](https://cockpit-project.org/running.html#installation--setup). (Fedora, RHEL, CoreOS, CentOS, Debian, Ubuntu, Clear Linux, Arch Linux, openSUSE Tumbleweed).

If you get the same problem with any of the supported distributions and Safari (if [the Safari workaround](https://cockpit-project.org/running/safari.html) doesn't work for you), then please open a new issue.

---

**Summary**: You're using a distribution specifically without systemd and Cockpit needs systemd to work. Use [a different distribution that provides systemd](https://cockpit-project.org/running.html#installation--setup).

**Comment by 0x30c4** [comment]
I'm using cockpit with nginx in-front of it and with SSL and the cockpit port is blocked by the firewall so it's not accessible from outside. 
Yes, I did installed cockpit to my debian based server. And while I failed to connect to it (because of the TLS error) with safari then I tried firefox on artix.

**Comment by garrett** [comment]
@0x30c4: You said you were using a Linux VM with Artix? OK, so you're _actually_ using some version of Cockpit on some version of Debian then? And you were using Artix just for a browser? If that's the case, please open a new issue with your full detailed setup, importantly including what specific distribution and version you're running Cockpit on and which version of Cockpit.

For an issue to be actionable, we need to know what specific distribution and version you're using Cockpit on, what version of Cockpit you're using, and your browser and version too.

Our new issue form covers these things. Please fill it out @ https://github.com/cockpit-project/cockpit/issues/new?labels=bug&template=bug_report.yml — Please be detailed. Without specifics, we cannot figure out what the problem is... and if there's a problem in Cockpit, then we wouldn't be able to fix it, as we'd need to know exactly what your setup is to be able to diagnose things.

This issue, which has been closed for a while, is _specifically_ about connecting to Cockpit with iOS, which doesn't even factor into what you've mentioned so far... so a new issue would make sense. Thanks!

---

Please note that if you're using Debian for your server, make sure you're using the latest version of Debian (which is Debian 11, also known as "bullseye") and you might want to consider using the backports too, which would let you have a more up-to-date version of Cockpit. Details @ https://cockpit-project.org/running.html#debian

**Comment by 0x30c4** [comment]
@garrett Done.
https://github.com/cockpit-project/cockpit/issues/17602
